base/operators.jl

```diff
@@ -153,10 +153,15 @@ const ÷ = div
 # mod returns in [0,y) or (y,0] (for negative y),
 # whereas mod1 returns in (0,y] or [y,0)
 mod1{T<:Real}(x::T, y::T) = (m=mod(x,y); ifelse(m==0, y, m))
+# mod with offset n
+mod{T<:Real}(x::T, y::T, n::T) = mod(x+y-n,y)+n
```

Comment:
Why is this `x+y-n` instead of just `x-n`? What if `n` is zero, or negative? Does this unnecessarily overflow for large `n`?